_posts/en/newsletters/2020-06-10-newsletter.md

```diff
@@ -0,0 +1,311 @@
+      described the attack almost three years ago and it's been
+      mentioned or alluded to by others on the [mailing list][lau
+      sighash2], in [BIPs][bip341 all amounts], and in Newsletters
+      [#11][news11 sighash] and [#46][news46 digest changes], among
+      other discussions.  Although the absence of any known exploits in
+      the past three years doesn't mean the attack isn't important, it
+      does point to a previous lack of urgency.  Perhaps it's the case
+      that any attacker who gains access to the software controller for
+      a hardware wallet is more likely to perform a different
+      attack---such as an [address substitution attack][hw
+      security]---that pays the attacker directly rather than overpaying
+      transaction fees.
+
+    - **Double signing can also lead to spending twice:** the attack
+      depends on using compromised software to get a hardware wallet
+      user to authorize two slightly different transactions (each with
+      two inputs) that look identical to the user.  Yet the same
+      compromised software could show the user two completely different
+      transactions (each spending a different input) that look
+      identical, resulting in paying the same recipient twice.  Both
+      attacks are indistinguishable from the user's perspective but the
+      fix for the fee overpayment attack doesn't fix the [spending twice
+      attack][maxwell spend twice].
+
+    - **Multisig setups may require multiple compromises:** to perform
+      the attack against funds secured by multisig with multiple
+      hardware wallets, each signer that is needed to meet the minimal
+      threshold (e.g. "2" in a "2-of-3" multisig) would need to be
+      tricked into signing the same two transaction variations.  For
+      thresholds that include an online wallet which knows the value of
+      the UTXO it's signing for (e.g. a policy-based remote signer), the
+      attack only works if that online wallet is also compromised.
+
+    A long-term solution to the attack is to change the transaction
+    digest so that each signature in a transaction commits to the value
+    of all UTXOs being spent in that transaction, making the signature
+    invalid if an attacker lies about the amount of any UTXO.  This was
+    [proposed][lau sighash2] by Johnson Lau in 2018 (see [Newsletter
+    #11][news11 sighash]) and has been included in the [BIP341][]
+    specification of [taproot][topic taproot] since its earliest public
+    drafts (see [Newsletter #46][news46 digest changes]).  If taproot is
+    adopted, then it should become perfectly safe for stateless signers
+    like hardware wallets to sign for taproot UTXOs without referencing
+    previous transactions.  Unfortunately, that still doesn't fix the
+    spending twice attack, which is an issue with the stateless design
+    of most hardware wallets that prevents them from internally tracking
+    their own transaction history.
+
```

Comment:
Should we also call out the software that is now broken such as BTCPay server and Wasabi if a user upgrades to the latest Trezor firmware? That is another drawback/tradeoff.

Reply:
We do this, we just don't explictly name the software that's broken.

> This has broken compatibility with several wallets that interface with Trezor devices, either directly or through HWI.